app/main/views.py

```diff
@@ -11,13 +11,18 @@ def index():
 
 @main.route('/get-resources')
 def get_resources():
-    resources = Resource.query.all()
+    resources = Resource.query.all() 
     resources_as_dicts = Resource.get_resources_as_dicts(resources)
     return json.dumps(resources_as_dicts)
 
 
 @main.route('/get-associations/<int:resource_id>')
 def get_associations(resource_id):
-    # TODO: write logic to retrieve and return associations for resource.
-    associations = []
+    pin_info = Resource.query.filter_by(id=resource_id).first()
```

Comment:
pls use `resource = Resource.query.get(resource_id)` instead; a slightly more "canonic" way. It's better to call it a resource, because for the purposes of back-end, you are dealing with a resource object, and the fact that it'll be used for a pin in the front end is not very relevant here.